modules/solid_mechanics/src/bcs/ExplicitDirichletBCBase.C

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExplicitMixedOrder.h"
+#include "ExplicitDirichletBCBase.h"
+#include "MooseError.h"
+#include "NonlinearSystemBase.h"
+#include "libmesh/numeric_vector.h"
+#include <iostream>
+#include <ostream>
+
+InputParameters
+ExplicitDirichletBCBase::validParams()
+{
+  InputParameters params = NodalBC::validParams();
+  return params;
+}
+
+ExplicitDirichletBCBase::ExplicitDirichletBCBase(const InputParameters & parameters)
+  : NodalBC(parameters),
+    _mass_diag(initLumpedMass()),
+    _u_old(_var.nodalValueOld()),
+    _u_dot_old(_var.nodalValueDotOld()),
+    _exp_integrator(
+        dynamic_cast<const ExplicitMixedOrder *>(&_sys.getTimeIntegrator(_var.number())))
+{
+  if (!_exp_integrator)
```

Comment:
```suggestion
  if (!_explicit_integrator)
```

exp sounds too much like exponential